src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestBindCallGen.generated.txt

```diff
@@ -0,0 +1,70 @@
+// <auto-generated/>
+#nullable enable
+
+using System.Linq;
+
+internal static class GeneratedConfigurationBinder
+{
+    internal static void Bind(this global::Microsoft.Extensions.Configuration.IConfiguration configuration, Program.MyClass obj) => BindCore(configuration, ref obj);
+
+    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref Program.MyClass obj)
+    {
+        if (obj is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(obj));
+        }
+        if (configuration.GetSection("MyString").Value is string stringValue0) { obj.MyString = stringValue0; }
```

Comment:
`configuration.GetSection(string)` always allocates a new `ConfigurationSection` object. I wonder if it would be better if we instead did:

```suggestion
        if (configuration["MyString"] is string stringValue0) { obj.MyString = stringValue0; }
```

I don't believe there is differing behavior (other than the extra alloc), but we should verify.